manager/cve_handler.py

```diff
@@ -344,35 +345,34 @@ def handle_get(cls, **kwargs):
         # check if advisory parameter comes with show_advisories parameter
         if 'advisory' in args and args['advisory'] and ('show_advisories' not in args or not args['show_advisories']):
             raise ApplicationException("Advisory parameter shouldn't be used without show_advisories parameter", 400)
-
         rh_account_id = get_account_id(connexion.context['user'])
-        asys_view = AffectedSystemsView(rh_account_id, synopsis, list_arguments, args, args, connexion.request.path, cls._ids_only)
+
+        system_advisories = {}
+        patch_access = None
+        if 'show_advisories' in args and args['show_advisories']:
+            # pylint: disable=singleton-comparison
+            raw_inv_ids = (SystemVulnerabilities.select(SystemPlatform.inventory_id)
+                           .join(SystemPlatform, on=(SystemVulnerabilities.system_id == SystemPlatform.id))
+                           .join(CveMetadata, on=(SystemVulnerabilities.cve_id == CveMetadata.id))
+                           .where(CveMetadata.cve == synopsis)
+                           .where(SystemVulnerabilities.rh_account_id == rh_account_id)
+                           .where((SystemPlatform.opt_out == False)  # noqa: E712
+                                  & (SystemPlatform.stale == False)  # noqa: E712
+                                  & (SystemPlatform.when_deleted.is_null(True)))).dicts()
+            inv_ids_list = [inv_id['inventory_id'] for inv_id in raw_inv_ids]
+            advisories_list = get_advisories_per_cve_from_db(synopsis)
+            system_advisories, patch_access = cls._get_advisories_per_system(advisories_list, inv_ids_list)
```

Comment:
Requesting all inventory IDs can cause performance issues when CVE is affecting thousand of systems - lot of data transferred over network and Patch need to process more data. I've tried to do some measurements and it seems the patch request takes ~0.5s more for CVE affecting ~1400 systems than requesting single page of 20 systems. And it will probably scale up for more systems.

We probably can't avoid requesting all inventory ids when we want to filter by advisory and get correct pagination. But let's at least minimize number of these all inventory ids requests.

- It's needed only when `args['advisory']` is set, otherwise it's sufficient to request only 1 page as we do now.
- We should do filtering of `advisories_list` before the Patch request. If no advisory will left in the list after filter, we can skip the Patch request.

Reply:
1. Yeah, it makes sense, thank you
2. We only could verify if advisories_list has filtered advisory or not and not send redundant request to patch if there is no filtered advisory in the list, but we still need to return all list of advisories for the CVE in case we want to get all possible advisories per system for this CVE, or the data after filtering will be inconsistent with data without filtering.